modules/frontend/traceid_sharder.go

```diff
@@ -53,10 +54,19 @@ func (s asyncTraceSharder) RoundTrip(pipelineRequest pipeline.Request) (pipeline
 
 	// execute requests
 	concurrentShards := uint(s.cfg.QueryShards)
+	// if concurrent shards is set, respect that value
 	if s.cfg.ConcurrentShards > 0 {
 		concurrentShards = uint(s.cfg.ConcurrentShards)
 	}
 
+	// concurrent_shards grater then query_shards should not be allowed because it would create
+	// more goroutines then the jobs to send these jobs to queriers.
+	if concurrentShards > uint(s.cfg.QueryShards) {
+		// set the concurrent shards to the total shards
+		concurrentShards = uint(s.cfg.QueryShards)
+		_ = level.Warn(s.logger).Log("concurrent_shards greater than query_shards is invalid, setting concurrent shards equal to query shards")
```

Comment:
warning may make more sense here than on every query:

https://github.com/grafana/tempo/blob/main/cmd/tempo/app/config.go#L141

Reply:
moved to config.go and added test.